internal/venice-common/src/main/java/com/linkedin/venice/pubsub/manager/TopicMetadataFetcher.java

```diff
@@ -150,22 +155,25 @@ class TopicMetadataFetcher implements Closeable {
   }
 
   // acquire the consumer from the pool
-  private PubSubConsumerAdapter acquireConsumer() {
+  PubSubConsumerAdapter acquireConsumer() {
     try {
-      return pubSubConsumerPool.take();
+      consumerWaitListSize.incrementAndGet();
+      long startTime = System.nanoTime();
+      PubSubConsumerAdapter consumerAdapter = pubSubConsumerPool.take();
```

Comment:
A safer approach from my understanding is to dynamically create new consumers if the pool doesn't have enough consumers since `consumer release` can fail.

Reply:
I agree but this hides bugs in code if any. And since consumers are expensive, due to leaks we might end up a lot of resources which could cause another issues. 

I believe we should not have any leaks after this fix. But even if we do we should be able to detect them via newly added metrics.